Documentation/v2.md

```diff
@@ -0,0 +1,48 @@
+# Dex v2
+
+## Streamlined deployments
+
+Many of the changes between v1 and v2 were aimed at making dex easier to deploy and manage, perhaps the biggest pain point for dex v1. Dex is now a single, scalable binary with a sole source of configuration. Many components which previously had to be set through the API, such as OAuth2 clients and IDP connectors can now be specified statically. The new architecture lacks a singleton component eliminating deployment ordering. There are no more special development modes; instructions for running dex on a workstation translate with minimal changes to a production system.
+
+All of this results in a much simpler deployment story. Write a config file, run the dex binary, and that's it.
+
+## More storage backends
+
+Dex's internal storage interface has been cleaned up and simplified, allowing dex to use more backing databases to store its state. A notable addition uses the Kubernetes API (a key value store) to store state, allowing dex to run natively on top of Kubernetes clusters.
+
+Dex v2 support Postgres, SQLite3 and the Kubernetes API, but we hope that starting with multiple backends helps ensure the storage interface is actually pluggable, rather than being coupled to one particular implementation.
+
+A more in depth discussion of existing storage options and how to add new ones can be found [here][storage-docs].
+
+## Additional improvements
+
+The rewrite came with several, miscellaneous improvements including:
+
+* More powerful connectors. For example the GitHub connector can now query for teams.
+* Combined the two APIs into a single [gRPC API][api-docs] with no complex authorization rules.
+* Expanded OAuth2 capabilities such as the implicit flow. 
+* Simplified codebase and improved testing.
+
+## Rethinking registration
+
+Dex v1 performed well when it could manage users. It provided features such as registration, email invites, password resets, administrative abilities, etc. However, login flows and APIs remain tightly coupled with concepts like registration and admin users even when v1 federated to an upstream identity provider (IDP) where it likely only had read only access.
+
+Many of v2's use cases focus on federation to other IPDs rather than managing users itself. Because of this, options associated with registration, such as SMTP credentials, have been removed. We hope to add registration and user management back into the project through orthogonal applications using the [gRPC API][api-docs], but in a way that doesn't impact other use cases.
+
+## Removed features
+
+It's important to note that dex v2 lacks certain features present in v1. For the most part __we aim at adding these features back into the project__, but in a way that installations have to opt into the additional configuration instead of making them on by default.
```

Comment:
~~It's important to note that~~ Dex v2 lacks many features present in v1.  *We aim to add most of these features back into v2* in a way that forces installations to *opt in* to a feature instead of *opt out* as it was before.